### PR TITLE
fix(serve): supplying help or version as an arg should throw

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
yup

**If relevant, did you update the documentation?**
N/A

**Summary**
- `webpack-cli serve help` should throw an error.
- Added a test case to ensure that `serve` respects `--no-color` flag.

**Does this PR introduce a breaking change?**
Nope

**Other information**
N/A

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
> Seems to work for me, can you give a failing case?

`webpack-cli serve help`

**Comment by anshumanv** [comment]
> webpack-cli serve help

It's a `command`, they already recognize args `--help`, not sure if we should allow user to write 2 commands.

/cc @webpack/cli-team

**Comment by evilebottnawi** [comment]
@anshumanv do you mean `webpack server --help`?

**Comment by anshumanv** [comment]
No @evilebottnawi, like `webpack serve help` -> 2 commands

We already support `webpack serve --help`

**Comment by snitin315** [comment]
Actually we support two help commands `webpack --help` and `webpack help` both works the same. Same is the case with `version`

**Comment by anshumanv** [comment]
> Actually we support two help commands webpack --help and webpack help both works the same. Same is the case with version

Different point, talking about whether we need 2 commands *simultaneously*

**Comment by snitin315** [comment]
It's really confusing and misleading. I think we should only support `--help` & `--version`. 
Hence we won't have to support `webpack serve help`.
`webpack serve --help` is sufficient.

**Comment by anshumanv** [comment]
> It's really confusing and misleading. I think we should only support --help & --version.
Hence we won't have to support webpack serve help.
webpack serve --help is sufficient.

+1 only creates confusion, /cc @evilebottnawi

**Comment by evilebottnawi** [comment]
Good question, I don't have the strong opinion here, but `git` support
```
git help fetch
git fetch --help
```

**Comment by anshumanv** [comment]
> Good question, I don't have the strong opinion here, but git supports

In this case `fetch` is the arg for the help command => only 1 command, we are supporting `serve help` to which is actually misleading

**Comment by evilebottnawi** [comment]
Yes we should remove `webpack serve help` supporting

**Comment by anshumanv** [comment]
@jamesgeorge007 can you update the PR to limit command count to 1?

**Comment by jamesgeorge007** [comment]
> Yes we should remove `webpack serve help` supporting

Couldn't agree more.

File: test/help/help-commands.test.js [hunk 1]
```diff
@@ -4,6 +4,11 @@ const { run } = require('../utils/test-utils');
 const helpHeader = 'The build tool for modern web applications';
 
 describe('commands help', () => {
+    it('throws error if supplied as an argument for subcommands', () => {
+        const { stderr } = run(__dirname, ['serve', 'help'], false);
```
Comment (anshumanv) [inline test/help/help-commands.test.js:8]: test for `help serve` also needed if we wish to support

**Pull request merged**